### PR TITLE
Set timeouts for urlopen calls

### PR DESCRIPTION
**Pull request by nelson-liu**
It's possible that `urlopen` hangs indefinitely, so we add a timeout to prevent this case (since exceptions are usually caught anyway)

### CODE REVIEW

File: codalab/lib/file_util.py [hunk 1]
```diff
@@ -62,7 +62,7 @@ def download_url(source_url, target_path, print_status=False):
     """
     Download the file at |source_url| and write it to |target_path|.
     """
-    in_file = urllib.request.urlopen(source_url)
+    in_file = urllib.request.urlopen(source_url, timeout=30)
```
Comment (percyliang) [inline codalab/lib/file_util.py]: Could we make this a constant that's used consistently throughout?

**Comment by nelson-liu** [comment]
i was talking to michael xie, and he saw that all of his workers went offline due to the power outage—however, his workers were running on the main (azure) instance, so ostensibly only the workers lost connectivity but not the server...I didn't think that the workers would freeze in that setting as well, but it seems like they do.

**Comment by nelson-liu** [comment]
Ok, this should be ready for another look

File: codalab/common.py [hunk 2]
```diff
@@ -10,6 +10,7 @@
 # http://semver.org/
 CODALAB_VERSION = '0.5.21'
 BINARY_PLACEHOLDER = '<binary>'
+URLOPEN_TIMEOUT = 30
```
Comment (teetone) [inline codalab/common.py]: Should we rename this to `URLOPEN_TIMEOUT_SECONDS`?

**Review by teetone — APPROVED** [review]

**Pull request merged**